x/evm/types/msg.go

```diff
@@ -147,37 +148,43 @@ func NewMsgEthereumTxContract(
 }
 
 func newMsgEthereumTx(
-	nonce uint64, to *ethcmn.Address, amount *big.Int,
+	nonce uint64, to *ethcmn.Address, amount *big.Int, // nolint: interfacer
 	gasLimit uint64, gasPrice *big.Int, payload []byte,
 ) MsgEthereumTx {
 	if len(payload) > 0 {
 		payload = ethcmn.CopyBytes(payload)
 	}
 
+	var recipient *Recipient
+	if to != nil {
+		recipient = &Recipient{Address: to.String()}
+	}
+
 	txData := TxData{
 		AccountNonce: nonce,
-		Recipient:    to,
+		Recipient:    recipient,
 		Payload:      payload,
 		GasLimit:     gasLimit,
-		Amount:       new(big.Int),
-		Price:        new(big.Int),
-		V:            new(big.Int),
-		R:            new(big.Int),
-		S:            new(big.Int),
+		Amount:       sdk.ZeroInt(),
+		Price:        sdk.ZeroInt(),
+		V:            []byte{},
+		R:            []byte{},
+		S:            []byte{},
```

Comment:
do these need to be big.Int here?

Reply:
type was migrated to bytes instead of `big.Int`

Reply:
sorry, still a bit confused about this. i see there being impl of both []byte and big.Int and was wondering why that was so,,

Reply:
Correct, that's for RLP encoding only because it needs to match the ethereum format